packages/js/components/src/media-uploader/media-uploader.tsx

```diff
@@ -96,7 +96,8 @@ export const MediaUploader = ( {
 						<MediaUploadComponent
 							onSelect={ onSelect }
 							allowedTypes={ allowedMediaTypes }
-							multiple={ multipleSelect }
+							// eslint-disable-next-line @typescript-eslint/no-explicit-any
+							multiple={ multipleSelect as any }
```

Comment:
Our [component MediaUploader](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/media-uploader/media-uploader.tsx#L43) and the [component MediaUpload](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor/src/components/media-upload#multiple) accept `multiple` as a `boolean`. However, they also accept strings. In this context, to achieve the desired behavior, we need to pass the string `add`.  This allows for selecting multiple items without pressing `shift` when clicking.

This is a [known issue](https://github.com/WordPress/gutenberg/issues/9137), and there [is a PR](https://github.com/WordPress/gutenberg/pull/53350) in `Gutenberg` to address it.

I've added an `eslint-disable` comment to bypass the TS error. We should remove it once [this PR](https://github.com/WordPress/gutenberg/pull/53350) is merged.

Reply:
I am ok with using `eslint-disable-next-line` for now, but for future reference it would be better to use: `// @ts-expect-error`, as this will actually fail the build when the types are fixed and allow us to remove this comment again.
The `eslint-disable` doesn't do that, which will make it harder for us to go back and update it once it has been fixed upstream.